pyproject.toml

```diff
@@ -176,54 +178,90 @@ ignore = [
 
 [tool.pixi.project]
 channels = ["conda-forge"]
-platforms = ["linux-64", "osx-arm64", "osx-64"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
 
+[tool.pixi.environments]
+# We resolve only two groups: cpu and gpu.
+# Then, multiple environments can be created from these groups.
+default = { solve-group = "cpugroup" }
+gpu = { features = ["gpu"], solve-group = "gpugroup" }
+tasks-cpu = { features = ["test", "examples"], solve-group = "cpugroup" }
+tasks-gpu = { features = ["test", "examples", "gpu"], solve-group = "gpugroup" }
+
+# ---------------
+# feature.default
+# ---------------
+
+# Dependencies from conda-forge.
 [tool.pixi.dependencies]
+#
+# Matching `project.dependencies`.
+#
 coloredlogs = "*"
 jax = "*"
-jax-dataclasses = "*"
-jaxopt = "*"
 jaxlib = "*"
 jaxlie = "*"
-lxml = "*"
-mediapy = "*"
-mujoco = "*"
-notebook = "*"
+jax-dataclasses = "*"
 pptree = "*"
+optax = "*"
 qpax = "*"
 rod = "*"
-sdformat14 = "*"
 typing_extensions = "*"
+#
+# Optional dependencies.
+#
+lxml = "*"
+mediapy = "*"
+mujoco = "*"
+scipy = "*"
+#
+# Additional dependencies.
+#
+pip = "*"
+
+# Dependendencies from PyPI.
+[tool.pixi.pypi-dependencies]
+jaxsim = { path = "./", editable = true }
+
+# ------------
+# feature.test
+# ------------
 
 [tool.pixi.feature.test.tasks]
-examples = { cmd = "jupyter notebook ./examples" }
 pipcheck = "pip check"
-test = { cmd = "pytest", depends_on = ["pipcheck"] }
+tests = { cmd = "pytest", depends_on = ["pipcheck"] }
 
 [tool.pixi.feature.test.dependencies]
 black = "24.*"
 idyntree = "*"
 isort = "*"
-pip = "*"
 pre-commit = "*"
 pytest = "*"
 pytest-icdiff = "*"
 robot_descriptions = "*"
 
+# ----------------
+# feature.examples
+# ----------------
+
+[tool.pixi.feature.examples.tasks]
+examples = { cmd = "jupyter notebook ./examples" }
+
+[tool.pixi.feature.examples.dependencies]
+notebook = "*"
+robot_descriptions = "*"
+
+# -----------
+# feature.gpu
+# -----------
+
 [tool.pixi.feature.gpu]
 platforms = ["linux-64"]
 system-requirements = { cuda = "12.1" }
 
 [tool.pixi.feature.gpu.dependencies]
-cuda-cupti = "*"
 cuda-version = "12.*"
-jaxlib = {version = "*", build = "*cuda*"}
-
-[tool.pixi.pypi-dependencies]
-jaxsim = { path = "./", editable = true }
-
-[tool.pixi.environments]
-default = { solve-group = "cpugroup" }
-gpu = { features = ["gpu"], solve-group = "gpugroup" }
-test-cpu = { features = ["test"], solve-group = "cpugroup" }
-test-gpu = { features = ["test", "gpu"], solve-group = "gpugroup" }
+cuda-nvcc-tools = "*"
+cuda-nvvm-tools = "*"
+cudnn = "9.3.0.*"
+jaxlib = { version = "<0.4.32", build = "*cuda*" }
```

Comment:
Probably it may be a good idea to track the reason of this constraint, i.e. add a link to https://github.com/conda-forge/jaxlib-feedstock/issues/285 in a comment?

Reply:
I agree, done in https://github.com/ami-iit/jaxsim/compare/ed7ac613af9fa5bf93b641f0c690765602c47cd0..a36408ed4002aad523a03a6b15dce75208526a31.